### ISSUE TITLE
The apikey field for the DB2 driver datasource does not support encryption

### ISSUE THREAD

**Issue by iamahern** [OpenLiberty/open-liberty#6505]
DB2 on Cloud supports the use of `Service IDs` which have associated `apiKey` fields for connecting to the DB:

https://console.bluemix.net/docs/services/Db2onCloud/iam.html#jdbc

This can be used in Liberty via:
```
  <properties.db2.jcc portNumber="50001" apikey="${SERVICE_ID_API_KEY}" databaseName="${DB_NAME}" serverName="${DB_URL}" sslConnection="true"/>
```

In testing we discovered that the `apikey` field does not support this field being encrypted (e.g. with `{aes}`). 

My best guess is that `apikey` needs to be added to this variable:
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/PropertyService.java#L101

**Comment by mswatosh** [comment]
What error messages are you seeing when you try to use an enycryped apikey?

**Comment by frowe** [comment]
I suspect there is no error, just the connection fails because we're not decrypting the apikey property.  PropertyService.isPassword() doesn't know about apikey.  I think the author is correct, we just need to add "apikey" to PASSWORD_PROPS in PropertyService.java

**Comment by njr-11** [comment]
I agree with the proposed change.  A couple of related changes are needed as well.  For properties with encryptable values, we should also be marking the metatype entry as ibm:type="password".  Oddly, enough, when I looked at the metatype for properties.db2.jcc, I don't see apiKey there at all, so it is likely we need to add the entry for it.

**Comment by iamahern** [comment]
SERVICE_ID_API_KEY is a variable I created in bootstrap.properties.

This code was tested 3 ways:
1. Plain text API key --> no problems
2. `{aes}` Bad credential issues
3. `{xor}` Bad credential issues.

I also decrypted the `{xor}` password to double check that I had not messed something else up in the encryption process.

**Comment by mswatosh** [comment]
Fixed by PR #6601
